src/package-install-scripts.js

```diff
@@ -280,15 +292,52 @@ export default class PackageInstallScripts {
 
     await Promise.all(workers);
 
-    // cache all build artifacts
-    for (const pkg of pkgs) {
-      if (this.packageCanBeInstalled(pkg)) {
-        const ref = pkg._reference;
-        invariant(ref, 'expected reference');
-        const loc = this.config.generateHardModulePath(ref);
-        const beforeFiles = beforeFilesMap.get(loc);
-        invariant(beforeFiles, 'files before installation should always be recorded');
-        await this.saveBuildArtifacts(loc, pkg, beforeFiles, set.spinners[0]);
+    // generate built package as prebuilt one for offline mirror
+    const offlineMirrorPath = this.config.getOfflineMirrorPath();
+    if (this.config.packBuiltPackages && offlineMirrorPath) {
+      for (const pkg of pkgs) {
+        if (this.packageCanBeInstalled(pkg)) {
+          let prebuiltPath = path.join(offlineMirrorPath, 'prebuilt');
+          if (!await fs.exists(prebuiltPath)) {
+            await fs.mkdirp(prebuiltPath);
+          }
+          const prebuiltFilename = getPlatformSpecificPackageFilename(pkg);
+          prebuiltPath = path.join(prebuiltPath, prebuiltFilename + '.tgz');
+          const ref = pkg._reference;
+          invariant(ref, 'expected reference');
+          const builtPackagePath = this.config.generateHardModulePath(ref);
+          const pkgConfig = await Config.create(
+            {
+              cwd: builtPackagePath,
+            },
+            this.reporter,
+          );
+          const stream = await pack(pkgConfig, builtPackagePath);
+
+          const hash = await new Promise((resolve, reject) => {
+            const validateStream = new crypto.HashStream();
+            stream
+              .pipe(validateStream)
+              .pipe(fs2.createWriteStream(prebuiltPath))
+              .on('error', reject)
+              .on('close', () => resolve(validateStream.getHash()));
+          });
+          pkg.prebuiltVariants = pkg.prebuiltVariants || {};
+          pkg.prebuiltVariants[prebuiltFilename] = hash;
+          // this.artifacts[`${pkg.name}@${pkg.version}`] = [prebuiltFilename];
```

Comment:
Can we remove this line if it isn't used?

Reply:
Doh, debugging code